libyul/AST.h

```diff
@@ -38,26 +38,28 @@ using Type = YulString;
 
 struct DebugData
 {
+	typedef typename std::shared_ptr<DebugData const> ConstPtr;
+
 	explicit DebugData(
-		langutil::SourceLocation _nativeLocation,
+		langutil::SourceLocation _nativeLocation = {},
 		langutil::SourceLocation _originLocation = {},
 		std::optional<int64_t> _astID = {}
 	):
 		nativeLocation(std::move(_nativeLocation)),
 		originLocation(std::move(_originLocation)),
-		astID(std::move(_astID))
+		astID(_astID)
 	{}
 
-	static std::shared_ptr<DebugData const> create(
+	static DebugData::ConstPtr create(
 		langutil::SourceLocation _nativeLocation = {},
 		langutil::SourceLocation _originLocation = {},
 		std::optional<int64_t> _astID = {}
 	)
```

Comment:
One micro optimization that's very non-invasive in comparison and that we could already do here would be to change this to
```suggestion
	static DebugData::ConstPtr create(
		langutil::SourceLocation _nativeLocation,
		langutil::SourceLocation _originLocation = {},
		std::optional<int64_t> _astID = {}
	)
```

And then to add a
```
static DebugData::ConstPtr create()
{
static DebugData::ConstPtr emptyDebugData = create({});
return emptyDebugData;
}
```

But we can also just leave all optimization tweaks like that for the future, both fine with me.